epochStart/metachain/economics.go

```diff
@@ -329,12 +346,91 @@ func (e *economics) adjustRewardsPerBlockWithLeaderPercentage(
 }
 
 // compute inflation rate from genesisTotalSupply and economics settings for that year
-func (e *economics) computeInflationRate(currentRound uint64, currentEpoch uint32) float64 {
+func (e *economics) computeInflationRateLegacy(currentRound uint64, currentEpoch uint32) (float64, error) {
 	roundsPerDay := common.ComputeRoundsPerDay(e.roundTime.TimeDuration(), e.enableEpochsHandler, currentEpoch)
+
 	roundsPerYear := numberOfDaysInYear * roundsPerDay
 	yearsIndex := uint32(currentRound/roundsPerYear) + 1
 
-	return e.rewardsHandler.MaxInflationRate(yearsIndex)
+	return e.rewardsHandler.MaxInflationRate(yearsIndex), nil
+}
+
+func (e *economics) computeInflationRate(currentRound uint64, currentEpoch uint32) (float64, error) {
```

Comment:
can you give as argument the previous epoch, the one you are computing the rewards for?